runtime/lua/vscode-neovim/api.lua

```diff
@@ -418,64 +418,4 @@ do
   end
 end
 
---------------------------
------- status item  ------
---------------------------
-
-do
-  local _items = {}
-  local status_items = setmetatable({}, {
-    __newindex = function(_, k, v)
-      _items[k] = v
-      local status = vim.tbl_values(_items)
-      status = vim.tbl_filter(function(i)
-        return i ~= ""
-      end, status)
-      M.action("refresh-status-items", { args = { status } })
-    end,
-    __index = function(_, k)
-      return _items[k]
-    end,
-  })
-
-  ---Creates a status item
-  ---
-  ---```lua
-  ---local test = vscode.get_status_item('test')
-  ---test.text = 'hello' -- Show the text
-  ---test.text = '' -- Hide the item
-  ---test.text = nil -- Close the item
-  ---test.text = '' -- error: The status item "test" has been closed
-  ---```
-  ---
-  ---@param id string The identifier of the item
-  function M.get_status_item(id)
```

Comment:
to help users migrate, we could leave this as an empty function for a couple releases, and show an info message.

```lua
function M.get_status_item()
    M.notify("Nvim statusline is now shown in vscode automatically. get_status_item was removed.")
end
```